Simplenote/Classes/SPNoteTableViewCell.swift

```diff
@@ -0,0 +1,248 @@
+    }
+
+    /// Autolayout Init
+    ///
+    func setupSubviews() {
+        previewTextView.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addSubview(previewTextView)
+
+        NSLayoutConstraint.activate([
+            previewTextView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+            previewTextView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+            previewTextView.topAnchor.constraint(equalTo: containerView.topAnchor),
+            previewTextView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
+            ])
+    }
+}
+
+
+// MARK: - Private Methods: Skinning
+//
+extension SPNoteTableViewCell {
+
+    /// Refreshes the current Style current style
+    ///
+    @objc
+    func refreshStyle() {
+        backgroundColor = Style.backgroundColor
+
+        let selectedView = UIView(frame: bounds)
+        selectedView.backgroundColor = Style.selectionColor
+        selectedBackgroundView = selectedView
+
+        previewTextView.interactiveTextStorage.tokens = [
+            SPHeadlineTokenName: [
+                .font: Style.headlineFont,
+                .foregroundColor: Style.headlineColor
+            ],
+            SPDefaultTokenName: [
+                .font: Style.previewFont,
+                .foregroundColor: Style.previewColor
+            ]
+        ]
+    }
+
+    /// Applies the TextView Insets, based on the accessoryStack's Width
+    ///
+    private func refreshTextViewInsets() {
+        let width = accessoryImageView.image?.size.width ?? CGFloat.zero
+        previewTextView.textContainerInset.right = width + Style.previewInsets.right
+    }
+}
+
+
+// MARK: - Cell Styles
+//
+private enum Style {
+
+    /// Preview's Text Insets
+    ///
+    static let previewInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 16)
+
+    /// Returns the Cell's Background Color
+    ///
+    static var backgroundColor: UIColor {
+        return .color(name: .backgroundColor)!
+    }
+
+    /// Headline Color: To be applied over the first preview line
+    ///
+    static var headlineColor: UIColor {
+        return .color(name: .noteHeadlineFontColor)!
+    }
+
+    /// Headline Font: To be applied over the first preview line
+    ///
+    static var headlineFont: UIFont {
+        return .preferredFont(forTextStyle: .headline)
+    }
+
+    /// Preview Color: To be applied over  the preview's body (everything minus the first line)
+    ///
+    static var previewColor: UIColor {
+        return .color(name: .noteBodyFontPreviewColor)!
+    }
+
+    /// Preview Font: To be applied over  the preview's body (everything minus the first line)
+    ///
+    static var previewFont: UIFont {
+        return .preferredFont(forTextStyle: .body)
+    }
+
+    /// Color to be applied over the cell upon selection
+    ///
+    static var selectionColor: UIColor {
+        return .color(name: .lightBlueColor)!
+    }
```

Comment:
Do you need to force unwrap the colors?

Reply:
I'm afraid we do, because most of these colors go inside a Dictionary, used to initialize a NSAtrributedString. And sending an optional color as `.foregroundColor` triggers a warning.

Now, that being said, we didn't need force unwrap **on every color**, but for consistency's sake...!!